cli/src/cmd/forge/verify/sourcify.rs

```diff
@@ -51,19 +31,37 @@ impl VerificationProvider for SourcifyVerificationProvider {
         let cache = project.read_cache_file()?;
         let (path, entry) = crate::cmd::get_cached_entry_by_name(&cache, &args.contract.name)?;
 
-        let path = args.contract.path.map_or(path, PathBuf::from);
+        if entry.solc_config.settings.metadata.is_none() {
+            eyre::bail!(
+                r#"Contract {} was compiled without the solc `metadata` setting.
+Sourcify requires contract metadata for verification.
+metadata output can be enabled via `extra_output = ["metadata"]` in `foundry.toml`"#,
+                args.contract.name
+            )
+        }
 
-        let mut files = HashMap::new();
+        let mut files = HashMap::with_capacity(2 + entry.imports.len());
 
-        let filename = path.file_name().unwrap().to_str().unwrap().to_owned();
-        let metadata_path =
-            config.out.join(&filename).join(format!("{}.metadata.json", args.contract.name));
+        let artifact_path = args.contract.path.map_or(path, PathBuf::from);
+        let artifact: ConfigurableContractArtifact = fs::read_json_file(&artifact_path)?;
```

Comment:
this is not the contract path rather than artifact path. currently, it tries reading `.sol` file as json

Reply:
right, fixed

Reply:
gg, testing now